keras_cv/layers/object_detection/multi_class_non_max_suppression_test.py

```diff
@@ -39,3 +39,57 @@ def test_decode_predictions_output_shapes(self):
         self.assertEqual(result["boxes"].shape, [8, 100, 4])
         self.assertEqual(result["classes"].shape, [8, 100])
         self.assertEqual(result["confidence"].shape, [8, 100])
+
+class XlaMlirBridgeNmsPredictionDecoderTest(tf.test.TestCase):
+    tf.config.experimental.enable_mlir_bridge()
+    @tf.function(jit_compile=True)
+    def test_decode_predictions_output_shapes(self):
+        classes = 10
+        predictions_shape = (8, 98208, 4 + classes)
+
+        predictions = tf.random.uniform(
+            shape=predictions_shape, minval=0.0, maxval=1.0, dtype=tf.float32
+        )
+        box_pred = predictions[..., :4]
+        confidence_pred = predictions[..., 4:]
+
+        layer = cv_layers.MultiClassNonMaxSuppression(
+            bounding_box_format="xyxy",
+            from_logits=True,
+            max_detections=100,
+        )
+
+        result = layer(box_prediction=box_pred, confidence_prediction=confidence_pred)
+
+        self.assertEqual(result["boxes"].shape, [8, 100, 4])
+        self.assertEqual(result["classes"].shape, [8, 100])
+        self.assertEqual(result["confidence"].shape, [8, 100])
+
+@unittest.expectedFailure
+class XlaNmsPredictionDecoderTest(tf.test.TestCase):
+    # TODO This is not failing as it seems if uncommented it will globally 
+    # disable the MLIR bridge. 
+    # Are we compiling in the same context?
+    # tf.config.experimental.disable_mlir_bridge()
```

Comment:
@smit-hinsu @paynecl Is there a way to control the MLIR bridge for a single test?

Reply:
Note I see other tests doing the equivalent of

```python
if (not tf_test_utils.is_mlir_bridge_enabled()):
  self.skipTest("TPU tests require MLIR bridge")
```

If this were running with both new bridge enabled and disabled then you'd test both paths without needing to disable yourself. But @paynecl is on rotation and would know best.

Reply:
We have also `@test_util.disable_mlir_bridge` but I don't that we have these symbols exported. The problem is more about if and how it is possible to control the bridge activation/disactivation for a single test.

Here I am more going to test/confirm the expected failure with the "old" bridge and the test pass with the new MLIR bridge as we are not sure that we could already enable the new bridge. But controlling the bridge in single test it is still useful for test coverage.